### PR TITLE
add new dict for dependencies

### PR DESCRIPTION
**Pull request by annacuomo**
I am not 100% sure of the syntax, could you check?

### CODE REVIEW

File: batch.py [hunk 1]
```diff
@@ -709,14 +709,15 @@ def manage_concurrency_for_job(job: hb.job.Job):
...
+    dependencies_dict = {}
     for gene in genes_of_interest:
         # final path for this gene - generate first (check syntax)
         plink_file = output_path(f'plink_files/{gene}')
         gene_dict[gene]['plink'] = plink_file
 
         # if the plink output exists, do not re-generate it
         if to_path(f'{plink_file}.bim').exists():
-            gene_dict[gene]['plink_job'] = None
+            dependencies_dict[gene]['plink_job'] = None
```
Comment (lgruen) [inline batch.py]:
```suggestion
```

File: batch.py [hunk 2]
```diff
@@ -734,7 +735,7 @@ def manage_concurrency_for_job(job: hb.job.Job):
             window_size=window_size,
             plink_file=plink_file,
         )
-        gene_dict[gene]['plink_job'] = plink_job
+        dependencies_dict[gene]['plink_job'] = plink_job
```
Comment (lgruen) [inline batch.py]:
```suggestion
        dependencies_dict[gene] = plink_job
```

File: batch.py [hunk 3]
```diff
@@ -777,7 +778,7 @@ def manage_concurrency_for_job(job: hb.job.Job):
             manage_concurrency_for_job(prepare_input_job)
             copy_common_env(prepare_input_job)
             if gene_dict[gene]['plink_job'] is not None:
-                prepare_input_job.depends_on(gene_dict[gene]['plink_job'])
+                prepare_input_job.depends_on(dependencies_dict[gene]['plink_job'])
```
Comment (lgruen) [inline batch.py]: GitHub doesn't let me make a multi-line suggestion here, but this should be:

```python
            dependency = dependencies_dict.get(gene)
            if dependency:
                prepare_input_job.depends_on(dependency)
```

**Review by lgruen — CHANGES_REQUESTED** [review]

File: batch.py [hunk 4]
```diff
@@ -709,14 +709,15 @@ def manage_concurrency_for_job(job: hb.job.Job):
 
     # for each gene, extract relevant variants (in window + with some annotation)
     # submit a job for each gene (export genotypes to plink)
+    dependencies_dict: Dict[str,hb.job.Job] = {}
```
Comment (lgruen) [inline batch.py]:
```suggestion
    dependencies_dict: Dict[str, hb.job.Job] = {}
```

File: batch.py (hunk 3 above)
Comment (annacuomo) [reply batch.py]: thanks!

File: batch.py [hunk 5]
```diff
@@ -15,6 +15,7 @@
 
 import click
 import logging
+from csv import Dict
```
Comment (MattWellie) [inline batch.py]:
```suggestion
```

for a few python versions now type hints don't need to be capitalised.

If you were to import a specific type here for the type hints, you'd want to use

```
from typing import Dict
```

File: batch.py [hunk 6]
```diff
@@ -709,14 +710,14 @@ def manage_concurrency_for_job(job: hb.job.Job):
 
     # for each gene, extract relevant variants (in window + with some annotation)
     # submit a job for each gene (export genotypes to plink)
+    dependencies_dict: Dict[str, hb.job.Job] = {}
```
Comment (MattWellie) [inline batch.py:713]:
```suggestion
    dependencies_dict: dict[str, hb.job.Job] = {}
```

File: batch.py [hunk 7]
```diff
@@ -776,8 +777,9 @@ def manage_concurrency_for_job(job: hb.job.Job):
             prepare_input_job = batch.new_python_job(f'Prepare inputs for: {gene}')
             manage_concurrency_for_job(prepare_input_job)
             copy_common_env(prepare_input_job)
-            if gene_dict[gene]['plink_job'] is not None:
-                prepare_input_job.depends_on(gene_dict[gene]['plink_job'])
+            dependency = dependencies_dict.get(gene)
+            if dependency:
+                prepare_input_job.depends_on(dependency)
```
Comment (MattWellie) [inline batch.py:782]: You could further condense this (but some people hate the syntax)

```suggestion
            if dependency := dependencies_dict.get(gene):
                prepare_input_job.depends_on(dependency)
```

**Review by MattWellie — APPROVED** [review]

File: batch.py (hunk 7 above)
Comment (lgruen) [reply batch.py:782]: Perfect! I always forget about the walrus ;)

**Review by lgruen — APPROVED** [review]

**Pull request merged**